go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -108,78 +114,75 @@ def update_go_mod(go_mod_content, dependencies)
             }
           end
 
-          SharedHelpers.run_helper_subprocess(
+          body = SharedHelpers.run_helper_subprocess(
             command: NativeHelpers.helper_path,
             env: ENVIRONMENT,
             function: "updateDependencyFile",
             args: { dependencies: deps }
           )
+
+          write_go_mod(body)
         end
 
-        def run_go_get(go_mod_content, go_sum)
-          File.write("go.mod", go_mod_content)
-          File.write("go.sum", go_sum.content) if go_sum
-          File.write("main.go", dummy_main_go)
+        def run_go_get
+          unless Dir.glob("*.go").any?
+            File.write("main.go", "package dummypkg\n")
+          end
 
           _, stderr, status = Open3.capture3(ENVIRONMENT, "go get -d")
           handle_subprocess_error(stderr) unless status.success?
-
-          updated_go_sum = go_sum ? File.read("go.sum") : nil
-          { go_mod: File.read("go.mod"), go_sum: updated_go_sum }
         end
 
         def parse_manifest_requirements(go_mod_content)
-          File.write("go.mod", go_mod_content)
+          SharedHelpers.in_a_temporary_directory do
```

Comment:
Nit: we can skip this temporary directory and IO.

AFAICT we read the file into ruby, then create a temporary directory and write the file back out there.
But since this doesn't change the base file, we could skip some IO by just parsing from the current directory?

Reply:
We pass in different versions of the `go.mod` file here, one before and one after the changes, for the `updated_reqs` we _could_ skip the IO, but we'd have to have to add a different helper I think?

Reply:
Ah we call it at different times, yeah that works 👍